storage/tree/node_id2.go

```diff
@@ -16,16 +16,35 @@ package tree
 
 import "fmt"
 
-// NodeID2 is a faster NodeID that does zero memory allocations in transforming
-// methods like Prefix and Sibling. NodeID2 can be used as a key in Golang maps
-// directly, as well as compared equal.
+// NodeID2 identifies a node of a Merkle tree. It is a bit string that counts
+// the node down from the tree root, i.e. 0 and 1 bits correspond to going to
+// the left or right child correspondingly.
```

Comment:
English speakers, does "correspond to ... correspondingly" sound ok?

Reply:
suggestion: `0 and 1 bits represent going to the left or right child correspondingly`.